### PR TITLE
Fix Website without connector panic log + Bring back sync chip for Websites

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

It's unclear to me why we were loading all connectors from the websites list. 
I removed it and removed the panic log checking for dangling websites, that was wrongly raising when we were deleting a website. 

Instead, I removed the panic log on the detail view. 
If there's an error when trying to fetch the connector, then we panic if not found, and raise a raw error: 

Screenshot 2024-09-19 at 13 34 11

It should never happen but if it does we want the user to get an error. 

## Risk

Can be rolled back. 

## Deploy Plan

Deploy front

### CODE REVIEW

File: front/pages/w/[wId]/builder/data-sources/[dsId]/index.tsx [hunk 1]
```diff
@@ -109,6 +109,21 @@ export const getServerSideProps = withDefaultUserAuthRequirements<{
     );
     if (connectorRes.isOk()) {
       connector = connectorRes.value;
+    } else {
```
Comment (spolu) [inline front/pages/w/[wId]/builder/data-sources/[dsId]/index.tsx:112]: We can remove the null `| null` above no?

**Review by spolu — APPROVED** [review]
LGTM

File: front/pages/w/[wId]/builder/data-sources/public-urls.tsx [hunk 2]
```diff
@@ -83,53 +75,13 @@ export const getServerSideProps = withDefaultUserAuthRequirements<{
...
-  const dataSources = removeNulls(dataSourcesWithConnector);
-
   return {
     props: {
       owner,
       subscription,
       plan,
       readOnly,
-      dataSources,
+      dataSources: websiteDataSources,
```
Comment (spolu) [inline front/pages/w/[wId]/builder/data-sources/public-urls.tsx:84]: Not sure I understand why we don't need the connector anymore, but trusting you on this one :+1:

File: front/pages/w/[wId]/builder/data-sources/[dsId]/index.tsx (hunk 1 above)
Comment (PopDaph) [reply front/pages/w/[wId]/builder/data-sources/[dsId]/index.tsx:112]: Nope, folders don't have a connectorId so no connector!

File: front/pages/w/[wId]/builder/data-sources/public-urls.tsx (hunk 2 above)
Comment (PopDaph) [reply front/pages/w/[wId]/builder/data-sources/public-urls.tsx:84]: Ok we needed that for the sync chip but it has been removed in August here: https://github.com/dust-tt/dust/pull/6601/files

And we were still loading connectors.

Comment (PopDaph) [reply front/pages/w/[wId]/builder/data-sources/public-urls.tsx]: Adding a commit to fix and bring back the sync chip.

**Review by JulesBelveze — APPROVED** [review]
Thanks @PopDaph 🙏🏼

**Pull request merged**